### PR TITLE
ghe support (host and api path prefix)

### PR DESCRIPTION
**Pull request by zxan1285**

### CODE REVIEW

**Comment by Verlic** [comment]
MD5: 2319ac34f4848755a639fd524038dfd3

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **21**
Affected Modules: **superagent, superagent, superagent, superagent, superagent, superagent, mime, mime, mime, mime, mime, superagent, superagent, superagent, superagent, uglify-js, uglify-js, uglify-js, uglify-js, debug, growl**
Summary:

- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Incorrect Handling of Non-Boolean Comparisons During Minification
- Incorrect Handling of Non-Boolean Comparisons During Minification
- Regular Expression Denial of Service
- Command Injection


## Detailed report
#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-extension-express-tools@1.1.0 > webtask-tools@3.2.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-extension-tools@1.2.1 > webtask-tools@2.2.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-oauth2-express@1.3.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > auth0-oauth2-express@1.3.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > auth0-extension-tools@0.14.0 > webtask-tools@2.2.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > webtask-tools@3.2.0 > superagent@1.8.5**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-github-deploy-extension@2.2.0 > auth0-extension-express-tools@1.1.0 > webtask-tools@3.2.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-github-deploy-extension@2.2.0 > auth0-extension-tools@1.2.1 > webtask-tools@2.2.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-github-deploy-extension@2.2.0 > auth0-oauth2-express@1.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > auth0-oauth2-express@1.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-github-deploy-extension@2.2.0 > webtask-tools@3.2.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-extension-express-tools@1.1.0 > auth0-extension-tools@1.2.1 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-extension-tools@1.2.1 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > auth0-extension-tools@0.14.0 > superagent@2.3.0**

---

#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-github-deploy-extension@2.2.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-github-deploy-extension@2.2.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-github-deploy-extension@2.2.0 > auth0-source-control-extension-tools@2.5.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Regular Expression Denial of Service

Affected Module: **debug**
Installed version: **2.6.8**
Patched versions: **>= 2.6.9 < 3.0.0 || >= 3.1.0**
Advisory: https://nodesecurity.io/advisories/534
Path: **auth0-github-deploy-extension@2.2.0 > mocha@3.5.3 > debug@2.6.8**

---

#### Command Injection

Affected Module: **growl**
Installed version: **1.9.2**
Patched versions: **>=1.10.2**
Advisory: https://nodesecurity.io/advisories/146
Path: **auth0-github-deploy-extension@2.2.0 > mocha@3.5.3 > growl@1.9.2**

---

File: server/lib/github.js [hunk 1]
```diff
@@ -94,7 +94,9 @@ const getTree = (repository, branch, sha) =>
   new Promise((resolve, reject) => {
     try {
       const github = new GitHubApi({
-        version: '3.0.0'
+        version: '3.0.0',
+        host: config('GITHUB_HOST') || 'api.github.com',
+        pathPrefix: config('GITHUB_API_PATH') || ''
```
Comment (dctoon) [inline server/lib/github.js]: Is the `/api/v3` fixed for all Github Enterprise deployments? If yes, can't we just add it to the `url` if the `GITHUB_HOST ` is set? So if `GITHUB_HOST` is set _and_ different from `api.github.com`, add the fixed url part `/api/v3` to the `url`.

File: webtask.json [hunk 2]
```diff
@@ -31,6 +31,14 @@
       "default": "master",
       "required": true
     },
+    "GITHUB_HOST": {
+      "description": "Host of the GitHub instance (should be api.github.com for GitHub)",
+      "default": "api.github.com",
+      "required": true
```
Comment (dctoon) [inline webtask.json]: I would suggest making this an optional config, if left empty fall back to `api.github.com`.

File: webtask.json [hunk 3]
```diff
@@ -31,6 +31,14 @@
       "default": "master",
       "required": true
     },
+    "GITHUB_HOST": {
+      "description": "Host of the GitHub instance (should be api.github.com for GitHub)",
```
Comment (dctoon) [inline webtask.json]: `Host of GitHub Enterprise instance, leave empty to use github.com`, WDYT?

**Comment by MrPrashantT** [comment]
Any ETA for this to be merged?

**Pull request merged**